### PR TITLE
GameDB: Various Fixes

### PR DESCRIPTION
**Pull request by forrvalhalla**
### Description of Changes
Adds GameDB fixes to the demo versions of Genki's racing titles, and renames said demos to match retail.

### Suggested Testing Steps
Make sure I didn't break the formatting lol. Tested the fixes and it looks good to me.

### CODE REVIEW

File: bin/resources/GameIndex.yaml [hunk 1]
```diff
@@ -29526,8 +29539,11 @@ SLPM-61120:
     mergeSprite: 1 # Align sprite fixes FMVs but not garbage in-game, so needs merge sprite instead.
     texturePreloading: 1 # Performs better with partial preload because it is slow on locations outside gameplay foremost.
 SLPM-61121:
-  name: "Kaidou - Touge no Densetsu"
+  name: "Kaido - Touge no Densetsu [Taikenban]"
   region: "NTSC-J"
+  gsHWFixes:
+    alignSprite: 1 # Fixes vertical lines.
+    wildArmsHack: 1 # De-blurs the 3D image.
```
Comment (RedDevilus) [inline bin/resources/GameIndex.yaml:29546]: image

File: bin/resources/GameIndex.yaml [hunk 2]
```diff
@@ -29490,8 +29500,11 @@ SLPM-61114:
   name: "Shadow of Rome"
   region: "NTSC-J"
 SLPM-61115:
-  name: "Racing Battle - C1 Grand Prix"
+  name: "Racing Battle - C1 Grand Prix [Taikenban]"
   region: "NTSC-J"
+  gsHWFixes:
+    alignSprite: 1 # Fixes vertical lines.
+    wildArmsHack: 1 # De-blurs the 3D image.
```
Comment (RedDevilus) [inline bin/resources/GameIndex.yaml:29507]: image

File: bin/resources/GameIndex.yaml [hunk 3]
```diff
@@ -28936,8 +28939,12 @@ SLPM-60205:
   name: "Initial D - Special Stage"
   region: "NTSC-J"
 SLPM-60206:
-  name: "Shutokou Battle 01"
+  name: "Shutokou Battle 01 [Taikenban]"
   region: "NTSC-J"
+  gsHWFixes:
+    alignSprite: 1 # Fixes vertical lines.
+    wildArmsHack: 1 # Improves visual clarity whilst upscaling.
+    roundSprite: 1 # Reduces graphics garbage on UI whilst upscaling.
```
Comment (RedDevilus) [inline bin/resources/GameIndex.yaml:28947]: Whitespaces in multiple locations:

image

File: bin/resources/GameIndex.yaml [hunk 4]
```diff
@@ -28908,8 +28908,11 @@ SLPM-60194:
   name: "Kotoba no Puzzle - Mojipittan"
   region: "NTSC-J"
 SLPM-60195:
-  name: "Kaidou Battle - Nikko, Haruna, Rokko, Hakone"
+  name: "Kaido Battle - Nikko, Haruna, Rokko, Hakone [Taikenban]"
   region: "NTSC-J"
+  gsHWFixes:
+    alignSprite: 1 # Fixes vertical lines.
+    wildArmsHack: 1 # De-blurs the 3D image.
```
Comment (RedDevilus) [inline bin/resources/GameIndex.yaml:28915]: Idem whitespaces on the last fix.

**Comment by forrvalhalla** [comment]
Oh gosh I totally screwed this huh, apologies.
I updated the PR, as long as the update actually went through everything should be golden. (Hopefully)

**Review by Mrlinkwii — APPROVED** [review]
all the  issues @RedDevilus pointed out are fixed

**Review by lightningterror — APPROVED** [review]

**Pull request merged**